doc/guide.rst

```diff
@@ -0,0 +1,179 @@
+
+Basic components
+----------------
+
+For every application the following parts are always required and a developer of an application
+will work with them.
+
+* The :class:`App <simpleline.App>` static class to initialize and run application.
+* The :class:`ScreenHandler <simpleline.render.screen_handler.ScreenHandler>` static class for
+  scheduling screens.
+* The :class:`UIScreen <simpleline.render.screen.UIScreen>` based classes to create screens which
+  will form the application.
+* :ref:`Widgets <widgets_label>` to show anything on the screens.
+* :ref:`Containers <containers_label>` to position widgets on the screen.
+
+It may look difficult to work with so many components but it is simple. Look on the next section to
+see how everything fits together.
+
+How to create simple application
+--------------------------------
+
+.. currentmodule:: simpleline.render.screen
+
+Interaction with a user is necessary to have a useful UI framework. To show anything to a user the
+:class:`UIScreen` class must be used. So we will subclass
+this class to create our screen and set title of the screen::
+
+    class DividerScreen(UIScreen):
+
+        def __init__(self):
+            # Set title of the screen.
+            super().__init__(title=u"Divider")
+            self._message = 0
+
+The ``self._message`` variable will be used later to show result to a user.
+
+The screens main purpose is to present content to a user. For this we need widgets and containers.
+
+The :attr:`UIScreen.window` attribute is the most important part of the screen for the rendering,
+it contains the :class:`WindowContainer <simpleline.render.containers.WindowContainer>`
+container which is created and filed up by the :meth:`UIScreen.refresh` method.
+Everything added to this container is printed to the monitor.
+So we should override the :meth:`UIScreen.refresh` method and call the parent's version to
+prepare the container. Then we can add :ref:`widgets <widgets_label>` to the container.
+So the screen will continue like this:
+
+.. code-block:: python
+    :dedent: 0
+
+        def refresh(self, args=None):
+            # Fill the self.window attribute by the WindowContainer and set screen title as header.
+            super().refresh()
+
+            widget = TextWidget("Result: " + str(self._message))
+            self.window.add_with_separator(widget)
+
+The :meth:`WindowContainer.add_with_separator() <simpleline.render.containers.WindowContainer.add_with_separator>`
+method will print blank line after the
+:class:`TextWidget's <simpleline.render.widgets.TextWidget>` text.
+
+Now a user have the header and result printed on the screen but it would be nice to give him/her a
+hint how to use the Divider screen. The best suited component of the screen for this is the
+:class:`Prompt <simpleline.render.prompt.Prompt>`.
+The :class:`Prompt <simpleline.render.prompt.Prompt>` class is responsible to guide user by giving
+him/her the possible options he/she can write.
```

Comment:
`him/her the possible options he/she can write.` -> `them the possible options to choose from.`